### DISCUSSION TITLE
请问vc-selection-indicator如何获取GeoJSON的properties信息？

### DISCUSSION THREAD

**Discussion by houluy** [zouyaoji/vue-cesium#222] [Q&A] [answered]
通过原生Cesium的Selection-Indicator点击加载的GeoJSON后能够在infoBox中显示`properties`各项信息，但是不知道使用<vc-selection-indicator>组件后，如何获取`properties`信息呢？目前通过`pickEvt`的参数`selectedFeature`打印`properties`是`underfined`。事实上只有`position`属性存在，其他`name`、`description`、`properties`均为`undefined`。

还有一个问题，我有多个GeoJSON文件，我利用<div v-for=xxx...创建了多个<VcDatasourceGeojson :data=xxx>子组件，这是否是正确的操作？效果上看加载速度非常慢，感觉有点问题。

谢谢！

**Comment by zouyaoji** [comment, marked as answer]
第一个问题:

pickEvt 事件的参数中可以拿到。` e.cesiumObject.properties`

第二个问题可以看看是网络耗时还是渲染耗时慢。

**Reply by houluy** [reply]
貌似不行，我组件是这么写的：

```vue
<template>
<vc-selection-indicator ref="selectionIndicationRef" @pick-evt="pickEvt"></vc-selection-indicator>
</template>

<script lang="ts" setup>
import { ref } from 'vue'
const pickEvt = (e) => {
  console.log(e.cesiumObject.properties)
}
</script>
```

然后报错了，如下：
![error](https://user-images.githubusercontent.com/11565649/178151941-5c05a1a0-b42b-419e-ae5c-11615c5ce112.png)

**Reply by houluy** [reply]
又测试了一下，发现点击GeoJSON位置会触发两次`pick-evt`，第一次pick了底图，返回了`e.cesiumObject = undefined`；第二次才pick了GeoJSON entity。

所以需要先判断一下`e.cesiumObject === undefined`？还是说通过什么方法让他不`pick`底图？

**Reply by zouyaoji** [reply]
判断一下

**Reply by houluy** [reply]
ok，感谢~
